dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -534,9 +549,10 @@ Spdp::data_received(const DataSubmessage& data, const ParameterList& plist)
     return;
   }
 
-  DCPS::MessageId msg_id = (data.inlineQos.length() && disposed(data.inlineQos)) ? DCPS::DISPOSE_INSTANCE : DCPS::SAMPLE_DATA;
-
-  handle_participant_data(msg_id, pdata);
+  handle_participant_data(
+    (data.inlineQos.length() && disposed(data.inlineQos)) ?  DCPS::DISPOSE_INSTANCE : DCPS::SAMPLE_DATA,
+    pdata,
+    DCPS::SequenceNumber(data.writerSN.high, data.writerSN.low));
```

Comment:
It is slightly more verbose, but I'd rather have this conversion from the RTPS high/low sequence numbers to `SeqeunceNumber` as it appears in `RtpsUdpDataLink::RtpsReader::process_data_i`.  I think this change (along with the note above) would allow `SequenceNumber.h` to remain unchanged in this PR.